src/shogun/regression/svr/MKLRegression.cpp

```diff
@@ -46,6 +47,10 @@ void CMKLRegression::init_training()
 {
 	ASSERT(m_labels && m_labels->get_num_labels())
 	ASSERT(svm)
-	ASSERT(svm->get_classifier_type() == CT_SVRLIGHT)
+#ifdef USE_SVMLIGHT
+	ASSERT(typeid(*svm) == typeid(CSVRLight))
```

Comment:
this is not gonna work... see the ref manual of c++ for details:
https://en.cppreference.com/w/cpp/language/typeid

in short use this
```
std::type_index(typeid(*svm)) == std::type_index(typeid(CSVRLight))
```

Reply:
done. should I send PR in the gpl submodule, or is this still not ready?

Reply:
yeah it'd be good to see how the CI react on this change